### ISSUE TITLE
Numpy dtypes flagged as errors in isinstance checks.

### ISSUE THREAD

**Issue by adampauls** [microsoft/pyright#5298]
**Describe the bug**
Pyright seems to be unhappy with numpy dtypes (like `np.uint8`) used in `isinstance` checks. 

**To Reproduce**
```python
import numpy as np

if isinstance(np.uint8(1), np.uint8):
    print("success")
else:
    print("failure")

if isinstance(np.uint16(1), np.uint8):
    print("failure")
else:
    print("success")
```

**Expected behavior**
Code should type check. It runs without error and prints `success` twice, indicating the `isinstance` checks are working as expected.

**VS Code extension or command-line**
1.1.314 from command line. 

**Additional context**
My guess is this will be one of those many cases where it is numpy's fault, but from the code I can't see why.

**Comment by erictraut** [comment]
The problem is that the bundled stubs that ship with numpy define `np.uint8` as a type alias to the type `unsignedinteger[_8Bit]`. This is a specialized generic class (i.e. a class that derives from `typing.Generic` whose type argument(s) have been provided using a subscript syntax). At runtime, specialized generic classes become `types.GenericAlias` instances, and `isinstance` generates an exception if you pass a `types.GenericAlias` instance as the second argument. This is why pyright is emitting an error here.

The underlying problem is that the numpy stubs are effectively lying to the type checker. In practice, the `unsignedinteger` type does not derive from `typing.Generic` and is not transformed into a `types.GenericAlias` instance when subscripted. Pyright doesn't know this because the stubs indicate otherwise.

You will probably need to suppress the error using a `# type: ignore` or `# pyright: ignore` comment in this case.

**Comment by adampauls** [comment]
Thanks, I see there is already a [bug](https://github.com/numpy/numpy/issues/23007) filed in numpy.
